src/main/scala/chisel3/simulator/package.scala

```diff
@@ -0,0 +1,96 @@
+package chisel3
+
+import svsim._
+import chisel3.reflect.DataMirror
+
+package object simulator {
+  implicit class SimulationController(controller: Simulation.Controller) {
+    def port(data: Data): Simulation.Port = {
+      val context = Simulator.dynamicSimulationContext.value.get
+      assert(context.controller == controller)
+      context.simulationPorts(data)
+    }
+  }
+
+  implicit class ChiselWorkspace(workspace: Workspace) {
+    def elaborateGeneratedModule[T <: RawModule](
+      generateModule: () => T
+    ): T = {
+      elaborateGeneratedModuleInternal(generateModule)._1
+    }
+    private[simulator] def elaborateGeneratedModuleInternal[T <: RawModule](
+      generateModule: () => T
+    ): (T, Seq[(Data, ModuleInfo.Port)]) = {
+      // Use CIRCT to generate SystemVerilog sources, and potentially additional artifacts
+      var someDut: Option[T] = None
+      val outputAnnotations = (new circt.stage.ChiselStage).execute(
+        Array("--target", "systemverilog", "--split-verilog"),
+        Seq(
+          chisel3.stage.ChiselGeneratorAnnotation { () =>
+            val dut = generateModule()
+            someDut = Some(dut)
+            dut
+          },
+          circt.stage.FirtoolOption("-disable-annotation-unknown"),
+          firrtl.options.TargetDirAnnotation(workspace.supportArtifactsPath)
+        )
+      )
+
+      // Move the relevant files over to primary-sources
+      val filelist =
+        new java.io.BufferedReader(new java.io.FileReader(s"${workspace.supportArtifactsPath}/filelist.f"))
+      try {
+        filelist.lines().forEach { immutableFilename =>
+          var filename = immutableFilename
+          /// Some files are provided as absolute paths
+          if (filename.startsWith(workspace.supportArtifactsPath)) {
+            filename = filename.substring(workspace.supportArtifactsPath.length + 1)
+          }
+          java.nio.file.Files.move(
+            java.nio.file.Paths.get(s"${workspace.supportArtifactsPath}/$filename"),
+            java.nio.file.Paths.get(s"${workspace.primarySourcesPath}/$filename")
+          )
+        }
```

Comment:
Why are we copying every file from the filelist instead of just using them where they are emitted by firtool?

Reply:
FWIW, typically `.f` files are used as the argument to list all of your Verilog source files to the simulator. I don't have a strong objection here, just seems a little weird.